nbstripout/_utils.py

```diff
@@ -101,10 +114,19 @@ def strip_output(nb, keep_output, keep_count, extra_keys=''):
         # Remove the outputs, unless directed otherwise
         if 'outputs' in cell:
 
-            # Default behavior strips outputs. With all outputs stripped,
-            # there are no counts to keep and keep_count is ignored.
             if not keep_output_this_cell:
-                cell['outputs'] = []
+                if not max_size:
+                    # Default behavior strips outputs. With all outputs stripped,
+                    # there are no counts to keep and keep_count is ignored.
+                    cell['outputs'] = []
+                else:
+                    # otherwise remove outputs larger than max_size
+                    for idx, output in enumerate(cell['outputs']):
+                        if get_size(output) > max_size:
+                            cell['outputs'][idx] = None
+                        elif 'execution_count' in output and not keep_count:
+                            output['execution_count'] = None
+                    cell['outputs'] = [out for out in cell['outputs'] if out is not None]
```

Comment:
The `if keep_output_this_cell and not keep_count` check won't apply because `keep_output_this_cell` is false if we've made it to this branch. So, for example, if a cell has multiple outputs and some of them are kept for being small enough, then we would still need to clear the count on those outputs. But you are right this could be simplified with your suggestion combined with removing `keep_output_this_cell` from the following check.